Merge 'tensorboard-integration-ku' with 'tensorboard-integration'

kadhirumasankar opened the pull request:
Added a function to `logger.py` which logs metrics (namely model score, the best model score, EMA score, and the best EMA score) to TensorBoard files inside `~/runs/uav-classifier/`

TensorBoard can be run using `tensorboard --logdir=~/runs/uav-classifier/`

Commit:
Added a Log.metric() function which logs `most_recent_model_score` and `model_highest_score` to tensorboard
TODO: The tensorboard writer doesn't get closed. Is that an issue?

bprimal22 committed:
Tensorboard Integrated

Commit:
Added tensorboard to requirements.txt

Commit:
Also logging ema_score and best_ema_score
Changed most_recent_model_score, model_high_score, new_ema_highest_score, ema_high_score to model_score, best_model_score, ema_score, best_ema_score respectively

bprimal22 committed:
added runs from tensorboard log

bprimal22 committed:
Tensorboard Integration

Commit:
tensorboard integration

Commit:
added ema to tensorboard logging and fixed where the runs save

bprimal22 committed:
fixed log_dir

kadhirumasankar committed:
Merge branch 'tensorboard-integration-ku' into tensorboard-integration-ap

Commit:
implented requested changes i.e. deleted a few lines of code

Commit:
Resolved conflicts with tensorboard-integration-ku

bprimal22 committed:
/runs taken out

alexwitt23 committed:
Merge pull request #8 from uavaustin/tensorboard-intgration-bp

Merge Tensorboard-intgration-bp with tensorboard-integration-ku

alexwitt23 commented on train/train_utils/logger.py:
```diff
@@ -2,6 +2,7 @@
 
 import pathlib
 import logging
+from torch.utils.tensorboard import SummaryWriter
```
See [here](https://google.github.io/styleguide/pyguide.html#22-imports) on imports. Only import packages and modules.

alexwitt23 commented on train/train_utils/logger.py:
```diff
@@ -35,4 +37,6 @@ def warning(self, message: str) -> None:
 
-    # def metric
+    # logs metric to tensorboard
+    def metric(self, tag, value, epoch):
```
`def metric(self, tag: str, value: float, epoch: int) -> None:`

alexwitt23 reviewed (approved):
lgtm

kadhirumasankar committed:
Merge branch 'tensorboard-integration-ku' into tensorboard-integration-ap

kadhirumasankar committed:
Merge pull request #7 from uavaustin/tensorboard-integration-ap

merge tensorboard-integration-ap with tensorboard-integration-ku

Commit:
Implemented Alex's suggestions
Changed import to align with Google styleguide
Type checking function arguments

Commit:
Training loss is now logged to TensorBoard
